lib/OpenQA/Utils.pm

```diff
@@ -127,12 +127,17 @@ sub save_base64_png($$$) {
     return $newfile;
 }
 
-sub image_md5_filename($) {
-    my ($md5) = @_;
+sub image_md5_filename($;$) {
+    my ($md5, $orig_fname) = @_;
+
+    my $ext;
+    $ext = 'txt' if defined $orig_fname && $orig_fname =~ /\.txt$/;
+    $ext = 'wav' if defined $orig_fname && $orig_fname =~ /\.wav$/;
+    $ext //= 'png';
 
     my $prefix = substr($md5, 0, 2);
     $md5 = substr($md5, 2);
-    return ($imagesdir . "/$prefix/$md5.png", $imagesdir . "/$prefix/.thumbs/$md5.png");
+    return ($imagesdir . "/$prefix/$md5.$ext", $imagesdir . "/$prefix/.thumbs/$md5.$ext");
```

Comment:
It's just *WRONG* - that's all. Why not save these in testresults ? they are very likely much much smaller - and the optipng reason does not apply to them.